### DISCUSSION TITLE
is their exclude folder options?

### DISCUSSION THREAD

**Discussion by kk-hiraskar** [pyinstaller/pyinstaller#6126] [PyInstaller] [answered]
In pyinstaller any option to exclude particular file or folder which is present inside module.
```
Example:
  matplotlib/mpl-data/fonts/       [~8MB]
  plotly/package_data/plotly.min.js
```

These are huge files which are in module, but actually not used/required for execution.

I searched on net, but not got any info for this optimization.
If any options available, kindly let me know.

**Comment by rokm** [comment, marked as answer]
There is no option to specify folders to exclude.

But you can always filter `Analysis.datas` in your .spec file to remove unwanted entries.

For example,

```python
# myprogram.spec

a = Analysis(...)

# Filter out all entries in matplotlib/mpl-data/fonts
mpl_font_path = os.path.join('matplotlib', 'mpl-data', 'fonts')
a.datas = [entry for entry in a.datas if not entry[0].startswith(mpl_font_path)]

pyz = PYZ(...)
exe = EXE(...)
coll = COLLECT(...)
```

**Reply by kk-hiraskar** [reply]
Thanks @rokm , this way it worked..

```py
a = Analysis( ... )

excludes = ['matplotlib\\mpl-data\\fonts', 'plotly\\package_data\\plotly.min.js', '\\sample_data', '\\images']
datas = []
for d in a.datas:
    skip=False
    for e in excludes:
        if e in d[0]:
            skip=True
            break
    if not skip: datas.append(d)
a.datas = TOC(datas)
```

**Now the final size reduced by 10 more MB.. Thanks..**
